src/orange/detail/BIHBuilder.cc

```diff
@@ -0,0 +1,222 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022-2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/detail/BIHBuilder.cc
+//---------------------------------------------------------------------------//
+#include "BIHBuilder.hh"
+
+#include "BoundingBoxUtils.hh"
+
+namespace celeritas
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Variadic-templated struct for overloaded operator() calls.
+ */
+template<typename... Ts>
+struct Overload : Ts...
+{
+    using Ts::operator()...;
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * "Deduction guide" for instantiating Overload objects w/o specifying types.
+ */
+template<class... Ts>
+Overload(Ts...) -> Overload<Ts...>;
+}  // namespace
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from vector of bounding boxes and storage for LocalVolumeIds.
+ */
+BIHBuilder::BIHBuilder(VecBBox bboxes,
+                       BIHBuilder::LVIStorage* lvi_storage,
+                       BIHBuilder::InnerNodeStorage* inner_node_storage,
+                       BIHBuilder::LeafNodeStorage* leaf_node_storage)
+    : bboxes_(bboxes)
+    , lvi_storage_(lvi_storage)
+    , inner_node_storage_(inner_node_storage)
+    , leaf_node_storage_(leaf_node_storage)
+{
+    CELER_EXPECT(!bboxes_.empty());
+
+    centers_.resize(bboxes_.size());
+    std::transform(bboxes_.begin(),
+                   bboxes_.end(),
+                   centers_.begin(),
+                   [&](BoundingBox const& bbox) { return center(bbox); });
```

Comment:
I think you can actually give it just the name of the function (unless it's overloaded, in which case you'd have to do a function pointer cast):
```suggestion
                   &celeritas::detail::center);
```